keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,995 @@
+node in memory. Its `Allocatable` field is used to keep track of the allocatable
+resources in memory. At the beginning of each scheduling cycle, scheduler takes
+a snapshot of all the nodes in the cluster, and updates their corresponding
+`NodeInfo`.
+
+For the scheduler with DRA enabled, right after taking the node snapshot, the
+scheduler also takes a snapshot of `DeviceClass`, and updates
+`NodeInfo.DynamicResources` if there is extended resource backed by DRA.
+
+For a node with extended resources from device plugin, its NodeInfo's
+Allocatable.ScalarResources is updated with the k8s `Node`'s object.
+For a node with extended resources backed by DRA, its NodeInfo's
+Allocatable.DynamicResources is updated based on DRA `DeviceClass` objects.
+
+The existing 'noderesources' plugin needs to be modified, such that a pod's
+extended resource request is checked against a NodeInfo's ScalarResources if the
+node uses device plugin, and checked against a NodeInfo's DynamicResources, if
+the request is for extended resources backed by DRA, then 'noderesources' plugin
+would pass and leave it to 'dynamicresource' plugin to check if it can be satisfied.
+
+The existing 'dynamicresources' plugin needs to be modified to account for the
+extended resource backed by DRA requests.
+
+#### EventsToRegister
+This registers all cluster events that might make an unschedulable pod schedulable,
+like finishing the allocation of a claim, or resource slice updates.
+
+The existing dynamicresurce plugin has registered all the events needed or
+extended reource backed by DRA, hence no change is needed.
+
+#### PreFilter
+It checks if the pod has any container requests for extended resources backed by DRA.
+If not, and no claims in the pod, then the plugin can return early, as there
+is nothing to do.
+
+If the pod still needs to be considered by the plugin, then it creates a
+special resource claim that has empty spec. The exact spec needs to be decided
+during Filter phase, as some node may have device plugin provide the capacity
+for the extended resource, some other node may have DRA provide the capacity.
+The requsts in the special resource claim need to vary for each node.
+
+#### Filter
+If a pod has extended resource backed by DRA, and the node does *not* have
+device plugin to provide the capacity for the resource, then the
+dynamicresource plugin needs to try allocate the resource by filling in the
+special claim's `Spec.Devices.Requests` field.
+
+One `request` is created per container, and per extended resource backed by DRA
+in the container. The `DeviceClass` in the request is randomly picked if there
+are multiple device classes advertising the extended resource. The `Name` of the
+request is determined by the container name and the extended resource name, in
+case the name conflicts with an existing device request, the next choice is
+deterministically picked until there is no conflict.
```

Comment:
What conflict? All request names in the ResourceClaim created for the extended resources are under control of the scheduler.

Reply:
there should not be conflict, removed it.